File: libs/js/src/gaiatest.ts
/**
 * gaiatest — Test helpers for starting an ephemeral Gaia daemon in JS/TS tests.
 *
 * STATUS: Stub — full implementation coming in a future release.
 *         Use the Go gaiatest package (libs/go/gaiatest) for integration tests today.
 *
 * Planned API:
 *
 * ```ts
 * import { GaiaTestServer } from '@stain-win/gaia-client/gaiatest';
 *
 * describe('my feature', () => {
 *   let server: GaiaTestServer;
 *
 *   beforeAll(async () => {
 *     server = await GaiaTestServer.start();
 *   });
 *
 *   afterAll(() => server.stop());
 *
 *   it('can read a secret', async () => {
 *     const client = server.createClient();
 *     const secret = await client.getSecret({ namespace: 'app', id: 'key' });
 *     expect(secret.value).toBe('value');
 *   });
 * });
 * ```
 *
 * Requirements (when implemented):
 *  - `gaia` binary in PATH or GAIA_BIN env var
 *  - Node.js 18+
 */

export interface GaiaTestServerOptions {
  /** Path to the gaia binary. Defaults to GAIA_BIN env var or PATH lookup. */
  bin?: string;
  /** Master passphrase for the ephemeral daemon. Defaults to 'gaiatest-dev'. */
  passphrase?: string;
}

/**
 * GaiaTestServer manages an ephemeral Gaia daemon subprocess for testing.
 * @stub Full implementation coming soon.
 */
export class GaiaTestServer {
  /** gRPC address of the running daemon, e.g. "127.0.0.1:50123". */
  readonly addr: string = '';
  /** Directory containing ca.crt, gaia_client.crt, gaia_client.key. */
  readonly certDir: string = '';

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  private constructor() {}

  /**
   * Start an ephemeral Gaia daemon.
   * @stub Not yet implemented — throws NotImplementedError.
   */
  static async start(_opts?: GaiaTestServerOptions): Promise<GaiaTestServer> {
    throw new Error(
      'GaiaTestServer.start() is not yet implemented. ' +
        'Use the Go gaiatest package (libs/go/gaiatest) for integration tests.',
    );
  }

  /**
   * Stop the daemon and clean up all temporary files.
   * @stub No-op until implemented.
   */
  async stop(): Promise<void> {
    // no-op stub
  }
}

File: libs/rust/src/gaiatest.rs
//! gaiatest — Test helpers for starting an ephemeral Gaia daemon in Rust tests.
//!
//! **STATUS: Stub** — full implementation coming in a future release.
//! Use the Go `gaiatest` package (`libs/go/gaiatest`) for integration tests today.
//!
//! # Planned API
//!
//! ```rust,ignore
//! use gaia_client::gaiatest::GaiaTestServer;
//!
//! #[tokio::test]
//! async fn test_my_feature() {
//!     let server = GaiaTestServer::start(Default::default()).await.unwrap();
//!     // server.addr  — gRPC address
//!     // server.cert_dir — directory with TLS certs
//!     server.stop().await;
//! }
//! ```
//!
//! # Requirements (when implemented)
//! - `gaia` binary in `PATH` or `GAIA_BIN` env var
//! - Tokio async runtime

/// Options for configuring the test server.
#[derive(Debug, Default)]
pub struct GaiaTestServerOptions {
    /// Path to the gaia binary. Defaults to `GAIA_BIN` env var or PATH lookup.
    pub bin: Option<String>,
    /// Master passphrase for the ephemeral daemon. Defaults to `"gaiatest-dev"`.
    pub passphrase: Option<String>,
}

/// An ephemeral Gaia daemon process for use in tests.
///
/// **Stub** — not yet functional.
pub struct GaiaTestServer {
    /// gRPC address of the running daemon, e.g. `"127.0.0.1:50123"`.
    pub addr: String,
    /// Directory containing `ca.crt`, `gaia_client.crt`, `gaia_client.key`.
    pub cert_dir: String,
}

impl GaiaTestServer {
    /// Start an ephemeral Gaia daemon for testing.
    ///
    /// # Errors
    /// Always returns `Err` until this stub is fully implemented.
    pub async fn start(_opts: GaiaTestServerOptions) -> Result<Self, String> {
        Err(
            "GaiaTestServer::start() is not yet implemented. \
             Use the Go gaiatest package (libs/go/gaiatest) for integration tests."
                .to_string(),
        )
    }

    /// Stop the daemon and clean up temporary files.
    ///
    /// No-op stub.
    pub async fn stop(self) {}
}
